### ISSUE TITLE
The body and content-length of cy.request is incorrect when binary data

### ISSUE THREAD

**Issue by wBingwei** [cypress-io/cypress#6343]
### Current behavior:

I'm using cy.request() to request a export URL for an Excel file, but the body of response is incorrect, it has encoded the binary data in some way. 

We can see the length of data is **33070** which is the same as I export from Chrome, but the response.body length is **32935** which is different.
![image](https://user-images.githubusercontent.com/15047949/73916236-08814400-48f8-11ea-9f38-82af053b2703.png)

And this is the expected data (open via Nodepad++):
![image](https://user-images.githubusercontent.com/15047949/73916319-32d30180-48f8-11ea-824e-98c7795a743d.png)
This is the wrong data:
![image](https://user-images.githubusercontent.com/15047949/73916349-3ebec380-48f8-11ea-8b93-c8f930e4c9ab.png)

### Desired behavior:

The response data length should be the same with the length in response header, and the data should be the same with exported Excel file.

### Test code to reproduce

The `url` is a URL which will return an excel
```js
cy.request(url).then((response)=>{
     debugger;
}）
```

### Versions
Cypress: 3.8.0
OS: Windows 10
Browser: Electron 78

**Comment by jennifer-shehane** [comment]
May be related to https://github.com/cypress-io/cypress/issues/6178

**Comment by bsafwen** [comment]
It really is a big disappointment for us and we had to drop Cypress because it does not support binary payload, why would the framework encode it? We are sending compressed binary data, and when we cannot decompress it because Cypress corrupted it.

**Comment by metalocator** [comment]
We also have binary files that fail to upload, but only when uploaded via Cypress.  Using the hex editor I was able to determine that the files were changed during (only) Cypress' upload.  We've tried updating attachFile options to no avail

```
cy.get('input#fileupload').attachFile(fixtureFile, {
        mimeType: "application/x-gzip",
    });
```


File 13 was uploaded by Cypress, 14 was done in the browser, notice the size change.
```
-rwxrwxr-x 1 vagrant vagrant 546275 Oct  6 15:22 'ml_156.sql (13).gz'
-rwxrwxr-x 1 vagrant vagrant 300597 Oct  6 15:56 'ml_156.sql (14).gz'
```

Length for the Cypress-uploaded file

`Content-Length: 546471`

For the correct file

`Content-Length: 300795`
